zuulcilint/__main__.py

```diff
@@ -82,21 +83,13 @@ def lint_playbook_paths(zuul_yaml_files: list[pathlib.Path]) -> list[str]:
 
 
 def get_all_zuul_yaml_files(files: list[str]) -> list[pathlib.Path]:
-    """Get all Zuul YAML files from the specified file(s) or path(s)."""
-    zuul_yaml_files = []
-    for file_path in files:
-        zuul_yaml_files.extend(zuul_utils.get_zuul_yaml_files(pathlib.Path(file_path)))
-    return zuul_yaml_files
-
+    """Get all Zuul YAML files(this includes both .yaml and .yml)."""
+    zuul_yaml_files = defaultdict(list)
+    for f in files:
+        for k, v in zuul_utils.get_zuul_yaml_files(pathlib.Path(f)).items():
```

Comment:
Consider renaming v to a name which indicates that it is a list of paths; and, perhaps, k (good_yaml & bad_yml) as well, depending on what you named this key in other places.